File: core/src/nl/t64/game/rpg/components/cutscene/CutsceneContainer.java
package nl.t64.game.rpg.components.cutscene;

import java.util.HashMap;
import java.util.Map;


public class CutsceneContainer {

    private final Map<String, Boolean> cutscenes;

    public CutsceneContainer() {
        this.cutscenes = new HashMap<>(1);
        this.loadCutscenes();
    }

    public boolean isPlayed(String cutsceneId) {
        return cutscenes.get(cutsceneId);
    }

    public void setPlayed(String cutsceneId) {
        cutscenes.put(cutsceneId, true);
    }

    private void loadCutscenes() {
        cutscenes.put(CutsceneId.SCENE_INTRO, false);
    }

}

File: core/src/nl/t64/game/rpg/components/cutscene/CutsceneContainer.kt
package nl.t64.game.rpg.components.cutscene


class CutsceneContainer {

    private val cutscenes: MutableMap<String, Boolean> = mutableMapOf(
        Pair(CutsceneId.SCENE_INTRO, false)
    )

    fun isPlayed(cutsceneId: String): Boolean {
        return cutscenes[cutsceneId]!!
    }

    fun setPlayed(cutsceneId: String) {
        cutscenes[cutsceneId] = true
    }

}
